Mathlib/Algebra/DirectLimit.lean

```diff
@@ -183,6 +184,43 @@ lemma lift_injective [IsDirected ι (· ≤ ·)]
   rw [lift_of] at hz
   rw [injective _ g hz, _root_.map_zero]
 
+variable (f) in
+/--
+For two family of modules `G` and `G'` indexed by the same set with direct system `f` and `f'`
+respectively, if there is a compatible family of linear-equivalence `eᵢ : Gᵢ ≅ Gᵢ'`, i.e. for
+`i ≤ j`, we have `eᵢ ∘ fᵢⱼ = fᵢⱼ' ∘ eⱼ`, there is a linear equivalence `lim G ≅ lim G'`.
+-/
+def congr [IsDirected ι (· ≤ ·)] (equiv : (i : ι) → G i ≃ₗ[R] G' i)
+    (compatible_equiv : ∀ i j h x, equiv j (f i j h x) = f' _ _ h (equiv _ x)) :
```

Comment:
I recommend writing this with composition so that the caller can use `ext`.

Reply:
Which version do you think is better`∀ i j h, equiv j ∘(f i j h) = (f' _ _ h) ∘ (equiv _)` or `∀ i j h, equiv j ∘ₗ (f i j h) = (f' _ _ h) ∘ₗ  (equiv _)`

Reply:
The second one is much better.

Reply:
(though you should write `∀ i j h, equiv j ∘ₗ f i j h = f' _ _ h ∘ₗ equiv _` without the parens)

Reply:
with linear map comp, the problem is we need to write `(equiv j).toLinearMap`

Reply:
Understood! Thanks

Reply:
Let me add the version for `AddCommGroups/Rings` etc in this file